gammapy/cube/psf_map.py

```diff
@@ -135,12 +157,25 @@ def geom(self):
     @classmethod
     def read(cls, filename, **kwargs):
         """Read a psf_map from file and create a PSFMap object"""
-        psfmap = Map.read(filename, **kwargs)
-        return cls(psfmap)
-
-    def write(self, *args, **kwargs):
+        with fits.open(filename, memmap=False) as hdulist:
+            psfmap = Map.from_hdulist(hdulist, "PSFMAP", "BANDSPSF", "auto")
+            try:
+                expmap = Map.from_hdulist(hdulist, "EXPMAP", "BANDSEXP", "auto")
+            except:
```

Comment:
Plain `except:` will catch any kind of error, and often silently swallows errors you didn't expect, leading to hard to debug / confusion by the user. Is an if statement checking if the expected key is in the hdulist possible? Or make the except more specific, but even catching KeyError isn't as nice as `if "EXPMAP" in hdulist" IMO.

Reply:
OK changed to ```if "EXPMAP" in hdulist```